examples/model_selection/plot_grid_search_stats.py

```diff
@@ -0,0 +1,568 @@
+"""
+==================================================
+Statistical comparison of models using grid search
+==================================================
+
+This example illustrates how to statistically compare the performance of models
+trained and evaluated using :class:`~sklearn.model_selection.GridSearchCV`.
+
+"""
+
+###############################################################################
+# We will start by simulating moon shaped data (where the ideal separation
+# between classes is non-linear), adding to it a moderate degree of noise.
+# Datapoints will belong to one of two possible classes represented
+# by two features. We will simulate 50 samples for each class:
+
+print(__doc__)
+import matplotlib.pyplot as plt
+import seaborn as sns
+from sklearn.datasets import make_moons
+
+X, y = make_moons(noise=0.352, random_state=1, n_samples=100)
+
+sns.scatterplot(
+    X[:, 0], X[:, 1], hue=y,
+    marker='o', s=25, edgecolor='k', legend=False
+).set_title("Data")
+plt.show()
+
+
+###############################################################################
+# We will compare the performance of SVC estimators that vary on their `kernel`
+# parameter, to decide which choice of this hyper-parameter predicts our
+# simulated data best. We will evaluate the performance of the models using
+# :class:`~sklearn.model_selection.RepeatedStratifiedKFold`, repeating 10 times
+# a 10-fold stratified cross validation using a different randomization of the
+# data in each repetition. The performance will be evaluated using
+# :class:`~sklearn.metrics.roc_auc_score`.
+
+from sklearn.model_selection import GridSearchCV, RepeatedStratifiedKFold
+from sklearn.svm import SVC
+
+param_grid = [
+    {'kernel': ['linear']},
+    {'kernel': ['poly'], 'degree': [2, 3]},
+    {'kernel': ['rbf']}
+]
+
+svc = SVC(random_state=0)
+
+cv = RepeatedStratifiedKFold(
+    n_splits=10, n_repeats=10, random_state=0
```

Comment:
I agree this would be better in practice, but for this example doing so changes the numbers that are mentioned at the bottom of the document. I could still change the state and manually change those numbers, but I guess that would be more prone to errors in the description.